### PR TITLE
Allow to change DICOM C-STORE SCP parameters

### PR DESCRIPTION
**Pull request by bastiao**
Allow to change parameters in DICOM C-STORE SCP server by changing java properties.

### CODE REVIEW

File: dicoogle/src/main/java/pt/ua/dicoogle/server/RSIStorage.java [hunk 1]
```diff
@@ -162,11 +165,11 @@ public RSIStorage(String [] Services, SOPList l)
...
-                nae2.setDimseRspTimeout(60000*300);
-                nae2.setIdleTimeout(60000*300);
-                nae2.setMaxPDULengthReceive(s.getMaxPDULengthReceive()+1000);
-                nae2.setMaxPDULengthSend(s.getMaxPDULenghtSend()+1000);
-                nae2.setRetrieveRspTimeout(60000*300);
+                nae2.setDimseRspTimeout(Integer.parseInt(System.getProperty("dicoogle.cstore.dimseRspTimeout", "18000000")));
+                nae2.setIdleTimeout(Integer.parseInt(System.getProperty("dicoogle.cstore.idleTimeout", "18000000")));
+                nae2.setMaxPDULengthReceive(s.getMaxPDULengthReceive()+Integer.parseInt(System.getProperty("dicoogle.cstore.appendMaxPDU", "1000")));
+                nae2.setRetrieveRspTimeout(Integer.parseInt(System.getProperty("dicoogle.cstore.retrieveRspTimeout", "18000000")));
+                nae2.setRetrieveRspTimeout(Integer.parseInt(System.getProperty("dicoogle.cstore.retrieveRspTimeout", "18000000")));
```
Comment (Enet4) [inline dicoogle/src/main/java/pt/ua/dicoogle/server/RSIStorage.java:172]: Duplicated lines.

```suggestion
                nae2.setRetrieveRspTimeout(Integer.parseInt(System.getProperty("dicoogle.cstore.retrieveRspTimeout", "18000000")));
```

**Review by Enet4 — CHANGES_REQUESTED** [review]
Nothing egregiously wrong here other than the duplicated lines. One should only take care when defining these Java properties (anything that's not a valid number could just make the storage service crash or show other problems). When we get to better integrate this with the server settings abstraction, we could consider enabling users to specify human readable durations (e.g. "3m", "5s", ...).

File: dicoogle/src/main/java/pt/ua/dicoogle/server/RSIStorage.java [hunk 2]
```diff
@@ -135,19 +135,22 @@ public RSIStorage(String [] Services, SOPList l)
...
-            this.nae.setDimseRspTimeout(60000*300);
-            this.nae.setIdleTimeout(60000*300);
-            this.nae.setMaxPDULengthReceive(s.getMaxPDULengthReceive()+1000);
-            this.nae.setMaxPDULengthSend(s.getMaxPDULenghtSend()+1000);
-            this.nae.setRetrieveRspTimeout(60000*300);
+            this.nae.setDimseRspTimeout(Integer.parseInt(System.getProperty("dicoogle.cstore.dimseRspTimeout", "18000000")));
+            //
+            this.nae.setIdleTimeout(Integer.parseInt(System.getProperty("dicoogle.cstore.idleTimeout", "18000000")));
+            this.nae.setMaxPDULengthReceive(s.getMaxPDULengthReceive()+Integer.parseInt(System.getProperty("dicoogle.cstore.appendMaxPDU", "1000")));
+            this.nae.setMaxPDULengthSend(s.getMaxPDULenghtSend()+Integer.parseInt(System.getProperty("dicoogle.cstore.appendMaxPDU", "1000")));
```
[the remark this answers is not in the document]
Comment (bastiao) [reply dicoogle/src/main/java/pt/ua/dicoogle/server/RSIStorage.java:152]: Yes, it was there already.

**Review by bastiao — COMMENTED** [review]
> Nothing egregiously wrong here other than the duplicated lines. One should only take care when defining these Java properties (anything that's not a valid number could just make the storage service crash or show other problems). When we get to better integrate this with the server settings abstraction, we could consider enabling users to specify human readable durations (e.g. "3m", "5s", ...).

Yes, agree. Considering that it is not easy merge in version 3, it need to be reimplemented in a different way.

**Review by Enet4 — APPROVED** [review]

**Pull request merged**